packages/app/src/systems/Core/components/CoinBalance.tsx

```diff
@@ -0,0 +1,55 @@
+import { forwardRef, useMemo } from "react";
+
+import { useBalances } from "../hooks";
+import { parseToFormattedNumber } from "../utils";
+
+import { Button, Tooltip } from "~/systems/UI";
+import type { Coin } from "~/types";
+
+export type CoinBalanceProps = {
+  coin?: Coin | null;
+  showMaxButton?: boolean;
+  onSetMaxBalance?: () => void;
+  gasFee?: bigint | null;
+};
+
+export const CoinBalance = forwardRef<HTMLDivElement, CoinBalanceProps>(
+  ({ coin, showMaxButton = true, onSetMaxBalance, gasFee }) => {
+    const { data: balances } = useBalances({ enabled: true });
+    const coinBalance = balances?.find(
+      (item) => item.assetId === coin?.assetId
+    );
+
+    const balance = useMemo(
+      () => parseToFormattedNumber(coinBalance?.amount || BigInt(0)),
+      [coinBalance?.assetId, coinBalance?.amount]
+    );
+
+    return (
+      <div className="flex items-center justify-end gap-2 mt-2 whitespace-nowrap">
```

Comment:
Since you created a `coin-balance.css` you can move all these tailwind helper classes for a unique class using `coinBalance` in the root `div` component and maybe creating a `coinBalance--balance` class also

Reply:
I think it's better to remove `coin-balance.css` and keep everything inline. will do that